packages/account-api/src/api/multichain/account.ts

```diff
@@ -0,0 +1,253 @@
+      });
+
+      this.#accounts.set(provider, accounts);
+      for (const id of accounts) {
+        this.#providersByAccountId.set(id, provider);
+      }
+    }
+  }
+
+  get id(): MultichainAccountId {
+    return this.#id;
+  }
+
+  get wallet(): MultichainAccountWallet<Account> {
+    return this.#wallet;
+  }
+
+  get index(): number {
+    return this.#index;
+  }
+
+  hasAccounts(): boolean {
+    // Use this map, cause if there's no accounts, then this map will also
+    // be empty.
+    return this.#providersByAccountId.size > 0;
+  }
+
+  getAccounts(): Account[] {
+    let allAccounts: Account[] = [];
+
+    for (const [provider, accounts] of this.#accounts.entries()) {
+      allAccounts = allAccounts.concat(
+        accounts.map((id) => provider.getAccount(id)),
+      );
+    }
+
+    return allAccounts;
+  }
+
+  getAccount(id: Account['id']): Account | undefined {
+    const provider = this.#providersByAccountId.get(id);
+
+    return provider?.getAccount(id);
+  }
+
+  get(selector: MultichainAccountSelector<Account>): Account | undefined {
+    const accounts = this.select(selector);
+
+    if (accounts.length > 1) {
+      throw new Error(
+        `Too many account candidates, expected 1, got: ${accounts.length}`,
+      );
+    }
+
+    if (accounts.length === 0) {
+      return undefined;
+    }
+
+    return accounts[0]; // This is safe, see checks above.
+  }
+
+  select(selector: MultichainAccountSelector<Account>): Account[] {
+    return this.getAccounts().filter((account) => {
+      let selected = true;
+
+      if (selector.id) {
+        selected &&= account.id === selector.id;
+      }
+      if (selector.address) {
+        selected &&= account.address === selector.address;
+      }
+      if (selector.type) {
+        selected &&= account.type === selector.type;
+      }
+      if (selector.methods !== undefined) {
+        selected &&= selector.methods.some((method) =>
+          account.methods.includes(method),
+        );
+      }
+      if (selector.scopes !== undefined) {
+        selected &&= selector.scopes.some((scope) => {
+          return (
+            // This will cover specific EVM EOA scopes as well.
+            isScopeEqualToAny(scope, account.scopes)
+          );
+        });
+      }
+
+      return selected;
+    });
+  }
+}
+
+/**
+ * Gets the multichain account ID from its multichain account wallet ID and its index.
+ *
+ * @param walletId - Multichain account wallet ID.
+ * @param groupIndex - Index of that multichain account.
+ * @returns The multichain account ID.
+ */
+export function toMultichainAccountId(
+  walletId: MultichainAccountWalletId,
+  groupIndex: number,
+): MultichainAccountId {
+  return `${walletId}/${groupIndex}`;
+}
+
+/**
+ * Gets the multichain account index from an account group ID.
+ *
+ * @param groupId - Account group ID.
+ * @returns The multichain account index if extractable, undefined otherwise.
+ */
+export function getGroupIndexFromAccountGroupId(
```

Comment:
Maybe this change since not all `AccountGroup` will have a group index.

```suggestion
export function getGroupIndexFromMultichainAccountId(
```